doc/firebase/firebase.md

```diff
@@ -0,0 +1,54 @@
+# Firebase
+
+## Real-time database vs Firestore
+
+**Firestore** seems to be the best fit because:
+- allows to store collections on documents instead of only a single large JSON tree
+- better offline capabilities
+- is also real-time
+- allows more advanced sorting (also [geo queries](#geo-queries))
+- scales better (if we get millions of users)
+
+https://firebase.google.com/docs/firestore/rtdb-vs-firestore
+![Real-time database vs Firestore](img/rtdb-firestore-comparison.png)
+
+## How Firestore stores data
+Firebase doesn't use a relational model, instead it only stores collections of documents.
+
+A document is very similar to JSON object, it supports arrays, nested objects, numbers, strings [and more](https://firebase.google.com/docs/firestore/manage-data/data-types?hl=en).
+
+## Queries possible with Firestore
+- Get a single document using its ID
+- Get all documents of a collection
+  - where clause
+  - order by clause
+
+Not a relational database so we cannot do joins, we have to manually fetch the referenced documents.
+
+All queries can either
+- be run once to get the data
+- be "observed" and get notified in real time when a document matching the query changes (will be useful for the chat feature)
+
+## Data model
+We will have an `items` collection containing item documents with the attributs defined below.
+
+![diagram](img/model.png)
+
+In the future we will also have a `users` collection and each item will have the userID of its creator.
```

Comment:
Do we already need to consider that maybe users want to "star" some items they're interested in? Like having a list of favorites. Would that change the model?

Reply:
I wasn't sure either if we should already think about the fields for future features so I only added the fields for the main features.
But you're right, it's important to consider it now to avoid having to change the model later.

For the list of favorites, I think we can simply add a `favorites` field to the user which contains an array of item's IDs.
I've just found out that we'll be able to retrieve all the items in the list of favorite using a [whereIn clause](https://firebase.google.com/docs/firestore/query-data/queries?hl=en#in_not-in_and_array-contains-any). I'll add that to the doc :)